schedulearn-frontend/src/components/Contexts/UsernameContext.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Component } from "react";
+
+interface UsernameProps {
+  children: React.ReactNode;
+}
+
+export interface UsernameState {
+  username: string;
+  setUsername: (username: string) => void;
+}
+
+export const UsernameContext = React.createContext<UsernameState>({
+  username: "not-from-state", 
```

Comment:
I recommend making this field nullable (ex: username?: string or later on user?:User), so that the login page could tell if a user is already logged in/so it would be possible to sign out.

Reply:
Field made nullable